src/diff/line.js

```diff
@@ -3,6 +3,11 @@ import {generateOptions} from '../util/params';
 
 export const lineDiff = new Diff();
 lineDiff.tokenize = function(value) {
+  if(this.options.stripTrailingCr) {
+    // remove all CR (\r) characters from the input string
+    value = value.replace(/\r+\n/g, '\n');
+  }
+
   let retLines = [],
       linesAndNewlines = value.split(/(\n|\r\n)/);
```

Comment:
This should probably just be `value.split("\n")` after this change, right?

Reply:
Well only if `stripTrailingCr===true`, to not be breaking. But even in that case, I'm not sure I understand the benefit?

Reply:
Ah, I was getting confused by https://github.com/kpdecker/jsdiff/issues/275#issue-546119016. After reading that issue, but not testing anything to confirm, I had wrongly believed that the `\n` appearing first in the pipe-separated list of alternatives `\n|\r\n/` meant that it would always match in preference to `\r\n`, making the entire regex equivalent to just `/(\n)/`, and that the logic using that regex had thus always been fundamentally broken. I thus *thought* that with my comment above I was just proposing cleaning up some misleading code that never really worked.

But some quick experimentation suggests that I'm wrong, at least in Node, Chrome, and Firefox!

```
> "foo\r\nbar\r\nbaz\r\n".split(/(\n)/)
[
  'foo\r', '\n',
  'bar\r', '\n',
  'baz\r', '\n',
  ''
]
```

I'm not sure, then, why @cctakaso thought, in https://github.com/kpdecker/jsdiff/issues/275, that reordering the `\r\n` and `\n` in the regex would fix anything. Is it possible there is some JavaScript environment out there, that @cctakaso was using, with a regex engine where the order of alternatives really does affect the result in the way that issue suggests? I would've thought this would be standardised and all implementations would follow the standard, but tbh I don't want to spend hours unpicking the meaning of the ECMAScript 3 standard where regexes got added to JavaScript to confirm that the behaviour here has always been unambiguously specified!

My new suggestion, then, is that I'd be in favour of reversing the order as @cctakaso suggested, to `/(\r\n|\n)/`, just to make absolutely sure there's no ambiguity and avoid any need to research this - even though I'm now like 85% sure the order doesn't matter and this won't actually have any effect!

Reply:
Oops, pasted the wrong thing. I obviously meant to paste this:

```
> "foo\r\nbar\r\nbaz\r\n".split(/(\n|\r\n)/)
[
  'foo', '\r\n',
  'bar', '\r\n',
  'baz', '\r\n',
  ''
]
```

Reply:
Yeah the problem #275 presents will be fixed by the option, but the solution they suggested wouldn't change much since the array that the regex produces would still contain all new lines as separate elements, and `'\n' !== '\r\n'`. 
I think there's just a misunderstanding and we shouldn't change the code for it.